packages/quantic/force-app/main/default/lwc/quanticUserActionsTimeline/__tests__/quanticUserActionsTimeline.test.js

```diff
@@ -0,0 +1,511 @@
+      await flushPromises();
+
+      const initializationError = element.shadowRoot.querySelector(
+        selectors.initializationError
+      );
+
+      expect(initializationError).not.toBeNull();
+    });
+  });
+
+  describe('controller initialization', () => {
+    it('should build the controller with the right parameters', async () => {
+      createTestComponent();
+      await flushPromises();
+
+      expect(functionsMocks.buildUserActions).toHaveBeenCalledTimes(1);
+      expect(functionsMocks.buildUserActions).toHaveBeenCalledWith(
+        exampleEngine,
+        {
+          options: {
+            ticketCreationDate: exampleTicketCreationDateTime,
+            excludedCustomActions: exampleExcludedCustomActions,
+          },
+        }
+      );
+    });
+
+    it('should call fetchUserActions with the right parameters', async () => {
+      createTestComponent();
+      await flushPromises();
+
+      expect(functionsMocks.fetchUserActions).toHaveBeenCalledTimes(1);
+      expect(functionsMocks.fetchUserActions).toHaveBeenCalledWith(
+        exampleUserId
+      );
+    });
+
+    it('should subscribe to state changes', async () => {
+      createTestComponent();
+      await flushPromises();
+
+      expect(functionsMocks.subscribe).toHaveBeenCalledTimes(1);
+    });
+  });
+
+  describe('when user actions data is found', () => {
+    describe('the following sessions section', () => {
+      describe('when following sessions are found', () => {
+        it('should display the following sessions section', async () => {
+          const element = createTestComponent();
+          await flushPromises();
+
+          const followingSessionsSection = element.shadowRoot.querySelector(
+            selectors.followingSessionsSection
+          );
+
+          expect(followingSessionsSection).not.toBeNull();
+        });
+
+        it('should display the show following sessions button', async () => {
+          const element = createTestComponent();
+          await flushPromises();
+
+          const toggleFollowingSessionsButton =
+            element.shadowRoot.querySelector(
+              selectors.toggleFollowingSessionsButton
+            );
+
+          expect(toggleFollowingSessionsButton).not.toBeNull();
+          expect(toggleFollowingSessionsButton.label).toBe(
+            'Show following sessions'
+          );
+        });
+
+        it('should not display the following sessions', async () => {
+          const element = createTestComponent();
+          await flushPromises();
+
+          const followingSessions = element.shadowRoot.querySelectorAll(
+            selectors.followingSessions
+          );
+
+          expect(followingSessions.length).toBe(0);
+        });
+
+        describe('toggling on and off the following sessions', () => {
+          it('should display and hide the following sessions when toggling on and off the show following sessions button', async () => {
```

Comment:
this test is a bit long, I would suggest to split it in 2 : toggling on & toggling off

Reply:
To be able to toggle off the user actions, they need be previously toggled on, so even if you split you will need to do the toggle on then toggle off in the toggling off test which will make the code even bigger.

Reply:
Maybe in terms of lines it would be a bit longer, but I think it would be more readable. Anyways up to you, not a deal breaker for me